### PR TITLE
Hacking guide

### PR DESCRIPTION
**Pull request by jfilak**

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+ABRT integration tests live in the test/runtest directory. The test are based
+on [BeakerLib](http://fedorahosted.org/beakerlib).
+
+Caution! It is not recommended to run the tests unless you really know what
+you are doing. Lot of tests changes configuration of the system they are
+running on and can be potentially harmful to the system upon failures.
+
+The tests are supposed to be run automatically on a solo virtual machine but it
+is possible tu run them on development machines. If you decide to update an
```
Comment (mhabrnal) [inline HACKING.md]: tu -> to

File: HACKING.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    $ ./testsuite
+
+If there is a crashing test and you want to run the test program under GDB, you
+can go to the directory with failing test (./testsuite.dir/[TEST NAME]) and use
+libtool to execute the test with gdb:
+
+    $ libtool --mode=execute gdb ./[TESTNAME]
+
+When creating a new test please use macros defined in
+tests/helpers/testsuite.h. The C complier is configured to include files form
```
Comment (mhabrnal) [inline HACKING.md]: compiler

**Comment by xsulca00** [comment]
Allow core_backtrace generating for root programs and for root-to-non-privileged programs:
`echo 2 > /proc/sys/fs/suid_dumpable`

**Comment by xsulca00** [comment]
Add examples how to validate generated core_backtrace for example:
https://github.com/abrt/abrt/blob/master/tests/runtests/ccpp-plugin/verify_core_backtrace.py

**Comment by xsulca00** [comment]
Enable ABRT log messages (debugging) for systemd journal (journalctl):
In `/etc/abrt/plugins/CCpp.conf` edit `VerboseLog = 3`

**Pull request merged**